frontend/src/features/admin-form/responses/ResponsesPage/storage/UnlockedResponses/ProgressModal/ProgressModal.tsx

```diff
@@ -6,42 +7,90 @@ import {
   UseDisclosureReturn,
 } from '@chakra-ui/react'
 
+import { XMotionBox } from '~templates/MotionBox'
+
+import { CanceledResult, DownloadResult } from '../../types'
+import { isCanceledResult } from '../../utils/typeguards'
+
+import { CompleteScreen } from './CompleteScreen'
 import { ProgressModalContent } from './ProgressModalContent'
 
 export interface ProgressModalProps
   extends Pick<UseDisclosureReturn, 'onClose' | 'isOpen'> {
   downloadPercentage: number
   isDownloading: boolean
   children: React.ReactNode
+  downloadMetadata?: DownloadResult | CanceledResult
+  onCancel: () => void
 }
 
+enum ProgressFlowStates {
+  Progress = 'progress',
+  Complete = 'complete',
+}
+
+const INITIAL_STEP_STATE: [ProgressFlowStates, 1 | -1] = [
```

Comment:
What do the `1|-1` mean again?

Reply:
the direction of motion: 1 is right, -1 is left